synapse/types/__init__.py

```diff
@@ -85,6 +85,9 @@
 # Collection[str] that does not include str itself; str being a Sequence[str]
 # is very misleading and results in bugs.
 StrCollection = Union[Tuple[str, ...], List[str], AbstractSet[str]]
+# Sequence[str] that does not include str itself; str being a Sequence[str]
+# is very misleading and results in bugs.
+StrSequence = Union[Tuple[str, ...], List[str]]
```

Comment:
`StrCollection` allows unordered sets of path components, which is definitely not what a path is. So I defined a `StrSequence` type.

Reply:
Adding a comment about unordered vs ordered to each of the types would be useful even it may be intrinsic to the name if you're familiar with Python.

Reply:
Done in 10e31dbb9fdd25881b776a2ca21e85f99a1cf905.